controller/syncid/id.go

```diff
@@ -0,0 +1,20 @@
+package syncid
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"github.com/argoproj/argo-cd/v3/util/rand"
+)
+
+var globalCount = &atomic.Uint64{}
+
+// Generate generates a new ID
+func Generate() (string, error) {
+	prefix := globalCount.Add(1)
```

Comment:
tiny part of me wonders if we should move this below the random string generation just to make it less likely we increment for a sync that never happens.